app/controllers/books_controller.rb

```diff
@@ -0,0 +1,48 @@
+class BooksController < ApplicationController
+  before_action :setbook, only: %i[show edit update destroy]
+  def index
+    @books = Book.all
+    render :index
+  end
+
+  def show
+  end
+
+  def new
+    @book = Book.new
+  end
+
+  def edit
+  end
+
+  def create
+   @book = Book.new(book_params)
+    if @book.save
+      redirect_to @book
+    else
+      redirect_to new_book_path(error: @book.errors.full_messages)
+    end
+  end
+
+  def update
+    if @book.update(book_params)
+      redirect_to @book
+    else
+      redirect_to edit_book_path(error: @book.errors.full_messages)
+    end
+  end
+
+  def destroy
+    @book.destroy
+    redirect_to root_path
+  end
+
+private
+
+  def setbook
+    @book = Book.find(params[:id])
+  end
```

Comment:
This one is like cache on method and when you call it, you can just use `book` instead of `@book`. But, in this case not sure that it works on view or not
```suggestion
  def book
    @book ||= Book.find(params[:id])
  end
```